### ISSUE TITLE
Unable to login on webpage

### ISSUE THREAD

**Issue by mnomansadiq** [engrjislam/SpringBootWebSecurity#1]
Can you please tell me how to add user into a users table? When i run this sample app it show a login page but unable to proceed further as there are not known credentials of either user/admin available in users collections.

**Comment by mnomansadiq** [comment]
ok, But these scripts seems to be a pure SQL and can't be use directly on monogoDB right? I am planning to test spring security with mongoDB only. Is there any mongoDB script ?

**Comment by engrjislam** [comment]
I think [this](https://github.com/engrjislam/SpringBootMongoDB) will help you a bit (since both are identical) ... If you try then you will get the solution.

**Comment by mnomansadiq** [comment]
But seem to be same as there is is no detail of how to add a new Admin/Users is it ?

**Comment by mnomansadiq** [comment]
I actually made a change related to that. Here is a method loadUserByUsername() in WebUserDetailsService looks like now:

 @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        try {

            User user = userRepository.findByUsername(username);
            if (user == null) {
                User newuser = new User();
                newuser.setEnabled(true);
                newuser.setUsername(username);
                newuser.setPassword(passwordEncoder().encode("123456"));

                Set<String> set = new HashSet<>();
                set.add("Admin");
                newuser.setRoles(set);
                userRepository.save(newuser);
                user = userRepository.findByUsername(username);
                
            }
            return new org.springframework.security.core.userdetails.User(user.getUsername(), user.getPassword(),
                    getAuthorities(user)
            );
        } catch (Exception e) {
            throw new UsernameNotFoundException("User not found");
        }
    }

It now add dynamically a Admin user if it not exist in DB. It seems to be working fine but i am now redirected to a page  which is showing access denied. Any idea about it ?

**Comment by mnomansadiq** [comment]
I got it thanks..

**Comment by engrjislam** [comment]
try with ROLE_ADMIN and / orROLE_USER

**Comment by mnomansadiq** [comment]
Yes this was a problem i fixed it.. Thanks

**Comment by engrjislam** [comment]
You can even write a better code in controller not directly in service like 
    
- load a form by "/signup" with GET request 
  * ask for `username` and `password`
- then catch these in the POST method (do what you did in loadUserByUsername method in this conversation and do not forget to set role ROLE_USER as default persmission)

best of luck

**Comment by mnomansadiq** [comment]
Yes offcourse this is a proper way to adding a new user through proper request from controller. This is just for my understanding and testing purpose. 
Will disturb you if not understand anything within this project
